src/hooks/useCart/useCart.test.ts

```diff
@@ -0,0 +1,415 @@
+          maxQuantity: 15
+        }
+      });
+    });
+
+    it("should maintain cart quantities when more ids are added", async () => {
+      expect.assertions(2);
+      mockFetch.mockReturnValueOnce({
+        then: () => mockQuotaResSingleId
+      });
+      let ids = ["ID1"];
+      const { rerender, result, waitForNextUpdate } = renderHook(() =>
+        useCart(ids, "CORRECT_KEY", "https://myendpoint.com")
+      );
+
+      await waitForNextUpdate();
+      await wait(() => result.current.updateCart("chocolate", 5));
+      expect(result.current.cart).toStrictEqual({
+        "toilet-paper": {
+          category: "toilet-paper",
+          quantity: 0,
+          maxQuantity: 2
+        },
+        chocolate: {
+          category: "chocolate",
+          quantity: 5,
+          maxQuantity: 15
+        }
+      });
+
+      mockFetch.mockReturnValueOnce({
+        then: () => mockQuotaResMultipleIds
+      });
+      ids = ["ID1", "ID2"];
+      rerender([ids, "CORRECT_KEY", "https://myendpoint.com"]);
+      await waitForNextUpdate();
+      expect(result.current.cart).toStrictEqual({
+        "toilet-paper": {
+          category: "toilet-paper",
+          quantity: 0,
+          maxQuantity: 4
+        },
+        chocolate: {
+          category: "chocolate",
+          quantity: 5,
+          maxQuantity: 30
+        }
+      });
+    });
+
+    it("should set error when updateCart is given a negative quantity", async () => {
+      expect.assertions(2);
+      mockFetch.mockReturnValueOnce({
+        then: () => mockQuotaResSingleId
+      });
+      const ids = ["ID1"];
+      const { result } = renderHook(() =>
+        useCart(ids, "CORRECT_KEY", "https://myendpoint.com")
+      );
+
+      await wait(() => {
+        result.current.updateCart("chocolate", -5);
+      });
+
+      expect(result.current.error?.message).toBe("Invalid quantity");
+      expect(result.current.cart).toStrictEqual({
+        "toilet-paper": {
+          category: "toilet-paper",
+          quantity: 0,
+          maxQuantity: 2
+        },
+        chocolate: {
+          category: "chocolate",
+          quantity: 0,
+          maxQuantity: 15
+        }
+      });
+    });
+
+    it("should set error when updateCart is given a quantity over the limit", async () => {
+      expect.assertions(2);
+      mockFetch.mockReturnValueOnce({
+        then: () => mockQuotaResSingleId
+      });
+      const ids = ["ID1"];
+      const { result } = renderHook(() =>
+        useCart(ids, "CORRECT_KEY", "https://myendpoint.com")
+      );
+
+      await wait(() => {
+        result.current.updateCart("chocolate", 100);
+      });
+      expect(result.current.error?.message).toBe("Insufficient quota");
+      expect(result.current.cart).toStrictEqual({
+        "toilet-paper": {
+          category: "toilet-paper",
+          quantity: 0,
+          maxQuantity: 2
+        },
+        chocolate: {
+          category: "chocolate",
+          quantity: 0,
```

Comment:
hmm for this right, from a user point of view, if theres max 15, and i put a 100,

does it make more sense to have `quantity: 0` or `quantity: 15`? The insufficient quota message should still be thrown haha

Reply:
Hmm, I prefer to do nothing if there are invalid values though. i.e. nothing is "committed" to the cart unless it's valid